nvbench/cuda_call.cuh

```diff
@@ -57,6 +57,21 @@
     }                                                                                              \
   } while (false)
 
+/// Throws a std::runtime_error if `call` doesn't return `cudaSuccess`.
+/// Resets the error with cudaGetLastError().
+#define NVBENCH_CUDA_CALL_RESET_ERROR(call)                                                        \
```

Comment:
Rather than add a new macro, let's add the `cudaGetLastError` call to the existing macros, since we'll always want to clear them.

Reply:
Deleted NVBENCH_CUDA_CALL_RESET_ERROR and moved cudaGetLastError() call to NVBENCH_CUDA_CALL macro.
Other macros don't need cudaGetLastError() because
1) Driver API calls don't set runtime last error
2) NVBENCH_CUDA_CALL_NOEXCEPT immediately exits the program.